packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionVector_def.hpp

```diff
@@ -0,0 +1,227 @@
+                         const int maxRegPerProc, ///< max number of regions per proc
+                         const std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > > rowMapPerGrp, ///< row maps in quasiRegion layout [in]
+                         const std::vector<RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> > > rowImportPerGrp, ///< row importer in region layout [in]
+                         const Xpetra::CombineMode combineMode ///< Combine mode for import/export [in]
+                         )
+{
+  /* Let's fake an ADD combine mode that also adds local values by
+   * 1. exporting quasiRegional vectors to auxiliary composite vectors (1 per group)
+   * 2. add all auxiliary vectors together
+   */
+#include "Xpetra_UseShortNames.hpp"
+  using Teuchos::TimeMonitor;
+
+  RCP<TimeMonitor> tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("regionalToComposite: 1 - compVec setup")));
+
+  const SC SC_ZERO = Teuchos::ScalarTraits<Scalar>::zero();
+
+  const size_t compVecLocalLength = compVec->getLocalLength();
+  compVec->putScalar(SC_ZERO);
+
+  tm = Teuchos::null;
+
+  {
+    RCP<Vector> quasiRegVec;
+    for(int grpIdx = 0; grpIdx < maxRegPerProc; ++grpIdx) {
+      tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("regionalToComposite: 2 - quasiRegVec")));
+      quasiRegVec = regVec[grpIdx];
+      TEUCHOS_ASSERT(Teuchos::nonnull(quasiRegVec));
+      quasiRegVec->replaceMap(rowImportPerGrp[grpIdx]->getTargetMap());
+
+      tm = Teuchos::null;
+      tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("regionalToComposite: 3 - partialCompVec")));
+
+      RCP<Vector> partialCompVec = VectorFactory::Build(rowImportPerGrp[0]->getSourceMap(), true);
+      TEUCHOS_ASSERT(Teuchos::nonnull(partialCompVec));
+      TEUCHOS_ASSERT(partialCompVec->getLocalLength() == compVecLocalLength);
+      // ToDo (mayr.mt) Use input variable 'combineMode'
+      partialCompVec->doExport(*quasiRegVec, *(rowImportPerGrp[grpIdx]), Xpetra::ADD);
+
+      tm = Teuchos::null;
+      tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("regionalToComposite: 4 - compVec->sumIntoLocalValue")));
+
+      ArrayRCP<const SC> partialCompVecData = partialCompVec->getData(0);
+      ArrayRCP<SC>       compVecData        = compVec->getDataNonConst(0);
+      for(size_t entryIdx = 0; entryIdx < compVecLocalLength; ++entryIdx) {
+        compVecData[entryIdx] += partialCompVecData[entryIdx];
+      }
+
+      tm = Teuchos::null;
+    }
+  }
+
+  return;
+} // regionalToComposite
+
+/*! \brief Sum region interface values
+ *
+ *  Sum values of interface GIDs using the underlying Export() routines. Technically, we perform the
+ *  exchange/summation of interface data by exporting a regional vector to the composite layout and
+ *  then immediately importing it back to the regional layout. The Export() involved when going to the
+ *  composite layout takes care of the summation of interface values.
+ */
```

Comment:
@mayrmt should we also pull-in `scaleInterfaceDOFs` as it is mainly vector based?

Reply:
I intended to move vector-only stuff, thought we should discuss this in our meeting.

Reply:
@lucbv Done.